src/Lunr/Core/Tests/ConfigurationLoadEnvironmentTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * This file contains the ConfigurationLoadFileTest class.
+ *
+ * SPDX-FileCopyrightText: Copyright 2011 M2mobi B.V., Amsterdam, The Netherlands
+ * SPDX-FileCopyrightText: Copyright 2022 Move Agency Group B.V., Zwolle, The Netherlands
+ * SPDX-License-Identifier: MIT
+ */
+
+namespace Lunr\Core\Tests;
+
+use PHPUnit\Framework\Attributes\BackupGlobals;
+
+/**
+ * This tests loading configuration files via the Configuration class.
+ *
+ * @covers \Lunr\Core\Configuration
+ */
+class ConfigurationLoadEnvironmentTest extends ConfigurationTestCase
+{
+
+    /**
+     * TestCase Constructor.
+     */
+    public function setUp(): void
+    {
+        $this->setUpArray($this->constructTestArray());
+    }
+
+    /**
+     * Test loading the environment correctly.
+     */
+    #[BackupGlobals(TRUE)]
+    public function testLoadEnvironment(): void
+    {
+        $_ENV = [];
+
+        $_ENV['TEST']     = 'Test1';
+        $_ENV['LOAD_ONE'] = 'Value';
+        $_ENV['LOAD_TWO'] = 'String';
+
+        $this->class->loadEnvironment();
+
+        $this->config['test'] = 'Test1';
+
+        $this->config['load']['one'] = 'Value';
+        $this->config['load']['two'] = 'String';
+
+        $this->assertEquals($this->config, $this->class->toArray());
+    }
+
+    /**
+     * Test loading the environment correctly.
+     */
+    #[BackupGlobals(TRUE)]
+    public function testLoadEnvironmentWithPrefix(): void
+    {
+        $_ENV = [];
+
+        $_ENV['TEST']                = 'Test1';
+        $_ENV['LUNR_LOAD_ONE']       = 'Value';
+        $_ENV['LUNR_LOAD_TWO']       = 'String';
+        $_ENV['LUNR_LOAD_THREE_TWO'] = 'String';
+
+        $this->class->loadEnvironment('Lunr');
+
+        $this->config['load']['one'] = 'Value';
+        $this->config['load']['two'] = 'String';
+
+        $this->config['load']['three']['two'] = 'String';
+
+        $this->assertEquals($this->config, $this->class->toArray());
+    }
+
+    /**
+     * Test loading a correct config file.
+     */
+    #[BackupGlobals(TRUE)]
+    public function testLoadFileOverwritesValues(): void
+    {
+        $_ENV = [];
+
+        $_ENV['TEST1']       = 'Overwrite Value';
+        $_ENV['TEST2_TEST3'] = 'Overwrite Array';
```

Comment:
I don't agree that loading `TEST2_TEST3` from the environment should unset `TEST2_TEST4` from the loaded config file. Overwriting the same value is fine, but unsetting unrelated values would be unexpected behavior.

This probably should verify as well that a `test4` is also not touched.

Reply:
Agreed, forgot to check that. The only exception is `TEST1=test` overwriting `$config['test1']['something']` IYAM.

Reply:
I can't get it to do this properly. It'll keep other keys just fine, but the nested ones don't seem to be merged correctly.